src/index.js

```diff
@@ -297,14 +297,15 @@ Multiaddr.prototype.equals = function equals (addr) {
  * // {family: 'IPv4', address: '127.0.0.1', port: '4001'}
  */
 Multiaddr.prototype.nodeAddress = function nodeAddress () {
-  if (!this.isThinWaistAddress()) {
-    throw new Error('Multiaddr must be "thin waist" address for nodeAddress.')
-  }
-
   const codes = this.protoCodes()
   const parts = this.toString().split('/').slice(1)
+
+  if (parts.length < 4) {
+    throw new Error('Multiaddr must be a valid address for nodeAddress.')
```

Comment:
This validation is not strict enough. For us to be able to generate a `nodeAddress` (which is `address` + `port` + `family`) we need to have at least `/ip4` or `/ip6` or `/dns4` or `/dns6` with `/tcp` or `/udp`. Simply checking if it has 4 parts  is not enough.